docs/notebook_tutorial/ocr.ipynb

```diff
@@ -0,0 +1,440 @@
vTab7fzzz3c6nTCxhoeH165du2nTpr179zLG/H4/0iHcbveNN964c+fObdu2hcPhJ554oqamZty4cSgGqqgoO3r02J/+9CfM09tvv/2WW24pLS3u7++fO3cuQmwDAwMIgM6bN6+0tBQ/PWPGDIfDQfo7fIYDXGLZbLasrAxWgdPpbGpqamtrW79+/YwZM+bMmZNOp9vb21tbWwcHB/v6+h555BFJkpqaml566SWPx7No0aLx48cPDAz09fUdOXIkkUgcOnTomWeeaWhoWLJkMSLjuJPr1q174oknNm7cmEqlxo0bN3HixOLiYqvVOjg4+MEHH7S2tr7xxhv79+93uVxFRUW4UfBLZbNZhOb1mnAXMuGMRuPUqVOvvfba/v7+gYGBN99887zzzlu5ciU4us/ni0bjhw4devXVV7Eoffe73503bx4cflQyQvFlUfko8MXB6US/JK1gCnsbCqYQ9d+1a1dnZ+cNN9xw1113VVZWqqra1tZ2+PDhhx566MiRI52dnQ899FBjY+P06dN/8IMfzJkzBxnlLS0tr7322ksvvRQKhT744IPVq1fffvvtWFhramoOHz78+9//fvv27Yyxs84664Ybbli8eHF5eTljLBaL7dmz5+mnn167dm1jY+ODDz7o8/lWrbqis7ObFLbUkWXVVqs1HA6jgv21114LhULxeHzJkiXf+c53QJKQZXz48OFXXnnlxRdf7O7ufvXVVy+55BKLxZLNZn0+HyKJWEHymj44PCvYC5966ql33303Ho8XFxffcMMNV1xxRVlZGcKpvb29zz333JNPPvnII49QNBCbLkQQUDGE8IrD4aBwAIzIfD6PcGE2m8X+h7UM2c3I3cE1Ii6Mun2PxzMwMLhz584HH3wQO9C11167YMGCefPmlZSUIKn88OHDzzzzzNtvvz04OPjnP/959uzZlzMGQUV4cZi28ZMzbMeOHb/73e+ee+45xti555571113LVmyRFGUcDiMhVsdWeXAtLzdq6666qKLLpIk6cUXX+zq6spkMmedddZXv/rV6urqQCDg8XgYY6lUyul0PvbYYy+88MLg4OCUKVOuu+66hQsXItlFr9f39PTs27fv17/+9cGDB1tbW//85z8jMOTz+WbMmAFvX2tr6549exoaGpBLjrOy2+3RaHT79u1dXV16vb6mpgYhGLPZjKD5/v37jUbj9OnT77zzzoULFzqdTkmS0uk0Nvj169e3tbU99NBDmUxm5syZBoOhvLwcZRAgdkhcw3hAgBIZ9HDakdZANBpNp9PxeDwYDPLZ1m1tbQjZ33jjjTfffHNlZSWetcfjwdbFt6YBvUMxQSqVev7553fv3p3L5S644ILrrrtu1qxZXq8XZ9LV1fX222+/+uqrzc3NGzZsmDJlSm1tLbzLuVzO5XIhL3vfvn0bNmyYOHHifffdt2DBAp1Oh6Y6u3fv/rd/+7empqZgMPjOO+9Mmzatrq4OnpKysrIjR448/fTTmzZt0ul06Ey1dOnSOXPmoBxhz549b7/99jPPPDM4OAiPnaRp64N7SVwuNrn3ZFk+44wzfvzjH1ut1o0bN/7sZz+D4/nee+9dvnx5KBRSVRVDBT7UF1544aWXXhoYGCgvL7/ppptWrVqF4gY4gCdMmPDQQw/l8/m1a9fOnDnz+uuv7+npMZlM06dP//73v//Nb36zv7//hRdemDhx4r333ouU/Ewm9+abbz777LOyLM+YMeO2227T6/XhcNTn891000133313f3//gw8++Pjjj/t8vuuuu27lypXIiQTDLmgA3aTSadQYmUwGeKxlWd6+fXsqlbrzzjuvuuqqhoaGRCLR19e3ZcuW5557bt++fS0tLffdd184HJ49e/Y3vvGNs88+G9XlXV1df/7zn19//fW+vr6Ojo6dO3eeccYZ4XDY7XbD9fXoo4++9957uVyuoaHhrrvuWrRoEX4RT+Spp55as2YNwq+VlZXnnnuu3W6nEmBJq0RRNO0uhFBLSkquvvrqaDT6y1/+MhQKPfzww/X19Q0NDbjqLVu2PProo8hPQBCDX6OIhsqago+AwBcEpw39YlqDlILW0gR+kWw2OzAw4HQ6r7jiinvuuae6uhqEo6KiYtasWUePHj169Ggymezt7a2qqvrRj36EQNXg4KDNZlu4cKHRaGxra3vrrbccDsf69etvuukmFBYZDIbXXnsNnrP58+ffeOONl156qd/vRyZ4VVVVWVlZOp2ORCJbt25tamqCTYZyKso6p7JEWZaR0JDL5bq7u3t6elwu19y5c++9994lSxajnQnySKqrq41GI/LMDh8+3NraWlpaSsIBPJ+jMjT42N9+++23334bhXtXXXXVl770pWnTpqGAC06aoqIij8fzxz/+EeEhm82GmiwcE0myyO0gAQ6oozHG0ul0cXGxw/FhWxJcnclkstttsFZxSrC8EXZEfV9ra+s777wD7jVv3ryrrrrqggsuCIfDfr9fVdWhoaFLL73U5/M1NzcjlveHP/zhcsZQKsEYg15DJpOJRCII2ezevfuxxx575ZVXJEm68sorb7nllvnz5yNdz2KxOJ1OSnyhfNu81scQ3judTufz+fC+w+EYN25ceXk5/gp1ooLWFAU291VXXVVRUaHT6aD2ic83NjZ+8MEHsVjsvffeu+GGG6qqqnK53OzZs2fNmtXd3R0IBHbt2rVo0aJkMknys0VFvgMHGrds2YIig2nTphUXF4MPvfzyy7hF06dP/9d//dc5c+YwxmKxmE6nq62tnTBhQl1dncfjefvtt3t7e7du3bpv374pU6aQ9EA0GoXkmMFgsFjMcCRQI0JQDQiTIncQ5aJmsxk54waDIZPJ2O12u91+xRVXfO9736uoKOvp6bPb7S6Xo79/EJcAlzNCcqiyVFU1nU4PDQ3FYrHS0tKSkpIlS5ZccsklCNIhqby6utpqtXZ2djY3NweDwdbW1kQi4fF4PB6P1WqBBBqy/hsaGn7wgx9cccUV
```

Comment:
Do we really need this checking here? I guess `OcrCharacterProcessor` has already ruled out all the empty text.